### PR TITLE
A bunch more doctests for string funcs

### PR DESCRIPTION
**Pull request by kshyatt**

### CODE REVIEW

**Review by fredrikekre — APPROVED** [review]
😄

File: base/strings/util.jl [hunk 1]
```diff
@@ -415,6 +415,18 @@ is a function, each occurrence is replaced with `r(s)` where `s` is the matched
...
+# Examples
+```jldoctest
+julia> replace("Python is a programming language.", "Python", "Julia")
+"Julia is a programming language."
+
+julia> replace("The quick foxes run quickly.", "quick", "slow", 1)
+"The slow foxes run quickly."
+
+julia> replace("The quick foxes run quickly.", "quick", "", 1)
+"The  foxes run quickly."
```
Comment (KristofferC) [inline base/strings/util.jl]: Double space here intended?

Comment (KristofferC) [inline base/strings/util.jl]: Yep..

**Review by KristofferC — APPROVED** [review]

**Comment by kshyatt** [comment]
Are the CI fails my fault?

**Comment by fredrikekre** [comment]
Nope, #23371 on bsd, spawn on mac and one timeout.

**Pull request merged**